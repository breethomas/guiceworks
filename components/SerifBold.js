// @flow
import React from 'react'
import css from '../styles/css'

css.fontFace({
  fontFamily: 'RNSCamelia-Bold',
  fontStyle: 'normal',
  fontWeight: 400,
  src: 'url("/static/fonts/RNSCamelia-Bold.otf") format("opentype")',
})

const style = css({
  fontFamily: 'RNSCamelia-Bold',
})

type Props = {
  className?: string,
  tag?: string,
}

const SerifBold = ({ className, tag, ...props }: Props) => {
  const Component = tag != null ? tag : 'span'
  return (
    <Component className={`${style} ${className || ''}`} {...props} />
  )
}

SerifBold.defaultProps = {
  className: '',
  tag: 'span',
}

export default SerifBold
