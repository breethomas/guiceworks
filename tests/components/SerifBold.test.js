import React from 'react'
import renderer from 'react-test-renderer'
import SerifBold from '../../components/SerifBold'

describe('SerifBold', () => {
  let component

  beforeEach(() => {
    component = (
      <SerifBold className="SerifBoldClassName" tag="p">
        SerifBold
      </SerifBold>
    )
  })

  it('shows the correct snapshot for the SerifBold component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
