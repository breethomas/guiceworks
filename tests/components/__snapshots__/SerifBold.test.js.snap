// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerifBold shows the correct snapshot for the SerifBold component tree 1`] = `
<p
  className="css-lcomo SerifBoldClassName"
>
  SerifBold
</p>
`;
